src/CabanaPD_Boundary.hpp

```diff
@@ -40,14 +55,53 @@ struct RegionBoundary
         , high_y( _high_y )
         , low_z( _low_z )
         , high_z( _high_z ){};
+
+    template <class PositionType>
+    KOKKOS_INLINE_FUNCTION bool inside( const PositionType& x,
+                                        const int pid ) const
+    {
+        return ( x( pid, 0 ) >= low_x && x( pid, 0 ) <= high_x &&
+                 x( pid, 1 ) >= low_y && x( pid, 1 ) <= high_y &&
+                 x( pid, 2 ) >= low_z && x( pid, 2 ) <= high_z );
+    }
+};
+
+// Define a subset of the system as the boundary with a cylinder.
+template <>
+struct RegionBoundary<Cylinder>
+{
+    double radius_in;
+    double radius_out;
+    double low_z;
+    double high_z;
+    double x_center = 0.0;
+    double y_center = 0.0;
+
+    RegionBoundary( const double _radius_in, const double _radius_out,
+                    const double _low_z, const double _high_z )
+        : radius_in( _radius_in )
+        , radius_out( _radius_out )
+        , low_z( _low_z )
+        , high_z( _high_z ){};
+
+    template <class PositionType>
+    KOKKOS_INLINE_FUNCTION bool inside( const PositionType& x,
+                                        const int pid ) const
+    {
+        double rsq = ( x( pid, 0 ) - x_center ) * ( x( pid, 0 ) - x_center ) +
```

Comment:
Would it be possible to pass a (general) user-defined function, instead of defining a set of pre-defining  geometries?

Reply:
Yes, that's possible but not something I'd thought to add. That would be possible with the new Custom version, but instead of passing that function here the user would just determine the particles on their own

Reply:
Added in the most recent commit